### PR TITLE
AB#58 Adds authorization UI components

### PR DESCRIPTION
**Pull request by kpjVideo**
Separates Header into a component.
Adds component for login, logout, and register buttons.
Adds loading page for use during auth0 grant init.

### CODE REVIEW

**Comment by kpjVideo** [comment]
Whoops, I appear to have merge conflicts here, let me try to resolve that first.

**Review by joenoon16510 — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -1,12 +1,14 @@
 import './App.css';
 import { BrowserRouter as Router, Route } from 'react-router-dom';
 import Landing from './pages/Landing';
+import Auth from './pages/Auth';
```
Comment (zlix12) [inline src/App.tsx:4]: This import causes the application to fail as `./pages/Auth` does not exist in this branch.

File: src/App.tsx [hunk 2]
```diff
@@ -1,12 +1,14 @@
 import './App.css';
 import { BrowserRouter as Router, Route } from 'react-router-dom';
```
Comment (zlix12) [inline src/App.tsx:2]: `react-router-dom` is also not listed in the project dependencies.

**Review by zlix12 — CHANGES_REQUESTED** [review]

**Review by joenoon16510 — APPROVED** [review]

**Review by zlix12 — APPROVED** [review]
LGTM

**Pull request merged**